cranelift-codegen/meta/src/gen_binemit.rs

```diff
@@ -18,10 +18,11 @@ fn gen_recipe(formats: &FormatRegistry, recipe: &EncodingRecipe, fmt: &mut Forma
     let inst_format = formats.get(recipe.format);
     let num_value_ops = inst_format.num_value_operands;
 
-    let want_args = recipe.operands_in.iter().any(|c| match c {
-        OperandConstraint::RegClass(_) | OperandConstraint::Stack(_) => true,
-        OperandConstraint::FixedReg(_) | OperandConstraint::TiedInput(_) => false,
-    });
+    let want_args = inst_format.has_value_list
```

Comment:
In the generated file, we don't have access to the instruction data's `args` when `want_args` is set to false. One of the new instructions added in this change, `savepoint`, needs to access `args` because it tells us the live variables at that point in the program. `savepoint` is a `MultiAry`instruction, but that is insufficient for `want_args` to be set to true since all other `MultiAry` instructions don't need `args`.

This change will generate more dead code than necessary since a lot of instructions have a value list but don't need access to `args`. The Rust compiler should be able to recognise and eliminate dead code, so it's not a major issue. Ideally, there is an alternative way that generates less dead code. Feel free to suggest an alternative or I can add a TODO comment above this change.

Reply:
Good to know, thanks; can you add a comment that it's for multi-ary instructions, please?